### DISCUSSION TITLE
0.23 migration failure

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#5894] [Q&A] [answered]
`2024/11/18 20:25:25 Failed to apply migration 1717233556_v0.23_migrate.go: migrated collection "boxsites" validation failure: indexes: (2: The index definition already exists..).`

```json
[
  {
    "id": "ktt8lrust0v0243",
    "name": "boxsites",
    "type": "base",
    "system": false,
    "schema": [
      {
        "system": false,
        "id": "4rlrfjtr",
        "name": "siteId",
        "type": "text",
        "required": true,
        "presentable": false,
        "unique": false,
        "options": {
          "min": 15,
          "max": 15,
          "pattern": ""
        }
      },
      {
        "system": false,
        "id": "mnmzpvxh",
        "name": "nameInt",
        "type": "text",
        "required": true,
        "presentable": false,
        "unique": false,
        "options": {
          "min": 3,
          "max": 15,
          "pattern": "^[a-z0-9]+$"
        }
      },
      {
        "system": false,
        "id": "ubnyba4b",
        "name": "nameExt",
        "type": "text",
        "required": true,
        "presentable": false,
        "unique": false,
        "options": {
          "min": 6,
          "max": 15,
          "pattern": "^[a-z0-9]+$"
        }
      },
      {
        "system": false,
        "id": "thairs4m",
        "name": "status",
        "type": "select",
        "required": true,
        "presentable": false,
        "unique": false,
        "options": {
          "maxSelect": 1,
          "values": [
            "active",
            "inactive",
            "error",
            "toggling",
            "creating",
            "moving",
            "deleting"
          ]
        }
      },
      {
        "system": false,
        "id": "co8w9efv",
        "name": "tier",
        "type": "text",
        "required": true,
        "presentable": false,
        "unique": false,
        "options": {
          "min": null,
          "max": null,
          "pattern": "^\\w+$"
        }
      }
    ],
    "indexes": [
      "CREATE INDEX `_ktt8lrust0v0243_created_idx` ON `boxsites` (`created`)",
      "CREATE UNIQUE INDEX `idx_unique_mnmzpvxh` ON `boxsites` (`nameExt`)",
      "CREATE UNIQUE INDEX `idx_unique_ubnyba4b` ON `boxsites` (`nameExt`)"
    ],
    "listRule": null,
    "viewRule": null,
    "createRule": null,
    "updateRule": null,
    "deleteRule": null,
    "options": {}
  }
]
```

**Comment by dschissler** [comment]
I just noticed that `nameExt` has two unique indexes on it.  Well, probably however that was allowed to happen should be addressed to not allow that (however it happened).  I'll try simply deleting one before doing the migration.  I don't know how it actually could have two at once, so one must be a dummy somehow.

**Reply by ganigeorgiev** [reply]
The duplicated indexes definition (+ indexes name check across all other collections) was introduced with v0.23.0. While technically I guess it is valid to have the same indexes on the same field it rarely makes sense in practice, hence the validator.

**Reply by dschissler** [reply]
Is that something that we're going to need to manually fix?  Well, I'm moving on from this because I have a deadline, it was an easy manual fix for me, and because I don't understand how or why I had two indexes in the first place, nor why it's no longer allowed.

**Reply by ganigeorgiev** [reply, marked as answer]
With v0.23.0 every `app.Save()` runs validation by default (and the checks are a lot more extensive too). This differ from <= v0.22.x where the `app.Dao()` write operations were essentially direct write to the DB meaning that it was possible either because of manual migration or some old bug in the UI to have resulted in the above state.

So, Yes, if the v0.23 migration fail for some reason the errors will have to be addressed manually. In your case dropping one of the duplicated index is the correct fix.
